src/Wavefront.OpenTracing.SDK.CSharp/WavefrontSpan.cs

```diff
@@ -35,7 +37,16 @@ public class WavefrontSpan : ISpan
             this.startTimestampUtc = startTimestampUtc;
             this.parents = parents;
             this.follows = follows;
-            this.tags = tags;
+
+            this.tags = (tags == null || tags.Count == 0) ?

```

Comment:
Handle the case where `this.tags` is set to null and setTagObject is called on the span. Basically initialize `this.tags` within setTagObject if it is null.